osbenchmark/builder/installers/opensearch_installer.py

```diff
@@ -0,0 +1,113 @@
+        self.host_cleaner = HostCleaner(self.path_manager)
+
+    def install(self, host, binaries, all_node_ips):
+        node = self._create_node()
+        self._prepare_node(host, node, binaries[OpenSearchInstaller.OPENSEARCH_BINARY_KEY], all_node_ips)
+
+        return node
+
+    def _create_node(self):
+        node_name = str(uuid.uuid4())
+        node_port = int(self.provision_config_instance.variables["node"]["port"])
+        node_root_dir = os.path.join(self.provision_config_instance.variables["test_execution_root"], node_name)
+        node_binary_path = os.path.join(node_root_dir, "install")
+        node_log_dir = os.path.join(node_root_dir, "logs", "server")
+        node_heap_dump_dir = os.path.join(node_root_dir, "heapdump")
+
+        return Node(name=node_name,
+                    port=node_port,
+                    pid=None,
+                    root_dir=node_root_dir,
+                    binary_path=node_binary_path,
+                    log_path=node_log_dir,
+                    heap_dump_path=node_heap_dump_dir,
+                    data_paths=None,
+                    telemetry=None)
+
+    def _prepare_node(self, host, node, binary, all_node_ips):
+        self._prepare_directories(host, node)
+        self._extract_opensearch(host, node, binary)
+        self._update_node_binary_path(node)
+        self._set_node_data_paths(node)
+        # we need to immediately delete the prebundled config files as plugins may copy their configuration during installation.
+        self._delete_prebundled_config_files(host, node)
+        self._prepare_config_files(host, node, all_node_ips)
+
+    def _prepare_directories(self, host, node):
+        directories_to_create = [node.binary_path, node.log_path, node.heap_dump_path]
+        for directory_to_create in directories_to_create:
+            self.path_manager.create_path(host, directory_to_create)
+
+    def _extract_opensearch(self, host, node, binary):
+        self.logger.info("Unzipping %s to %s", binary, node.binary_path)
+        self.executor.execute(host, "tar -xzvf {} --directory {}".format(binary, node.binary_path))
+
+    def _update_node_binary_path(self, node):
+        node.binary_path = os.path.join(node.binary_path, "opensearch*")
+
+    def _set_node_data_paths(self, node):
+        node.data_paths = [os.path.join(node.binary_path, "data")]
+
+    def _delete_prebundled_config_files(self, host, node):
+        config_path = os.path.join(node.binary_path, "config")
+        self.logger.info("Deleting pre-bundled OpenSearch configuration at [%s]", config_path)
+        self.path_manager.delete_path(host, config_path)
+
+    def _prepare_config_files(self, host, node, all_node_ips):
+        config_vars = self.get_config_vars(host, node, all_node_ips)
+        self.config_applier.apply_configs(host, node, self.provision_config_instance.config_paths, config_vars)
+
+    def get_config_vars(self, host, node, all_node_ips):
+        provisioner_defaults = {
```

Comment:
Nit: I doubt there's going to be another iteration of an installer but how do you feel about adding these keys of provisioner_defaults to an enum? 

Not sure where else provisioner_defaults might be used in the code. It might make this dictionary a more muddled; However, I'm curious if we should start implementing enums throughout OSB to enable reusability of terms and if we were ever to update the naming conventions of variables such as these, it'd be easier. Curious to what others think too.